tests/epics/signal/test_signals.py

```diff
@@ -43,41 +43,98 @@
+            self.__create_signal(protocol, Array1D[np.int16], "int16a")
+            self.__create_signal(protocol, Array1D[np.int32], "int32a")
+            self.__create_signal(protocol, Array1D[np.float32], "float32a")
+            self.__create_signal(protocol, Array1D[np.float64], "float64a")
+            self.__create_signal(protocol, Sequence[str], "stra")
+        self.__create_signal("pva", Array1D[np.int8], "int8a")
+        self.__create_signal("pva", Array1D[np.uint16], "uint16a")
+        self.__create_signal("pva", Array1D[np.uint32], "uint32a")
+        self.__create_signal("pva", Array1D[np.int64], "int64a")
+        self.__create_signal("pva", Array1D[np.uint64], "uint64a")
+        self.__create_signal("pva", MyTable, "table")
+        self.__create_signal("pva", Array1D[np.int64], "ntndarray:data")
+
+    def __create_signal(
+        self, protocol: Protocol, dtype: type[SignalDatatypeT], name: str
+    ):
+        signal = epics_signal_rw(dtype, f"{protocol}://{self._name}:{protocol}:{name}")
+        setattr(self, f"{protocol}_{name}".replace(":", "_"), signal)
+        self._all_signals[protocol][name] = signal
+
+    def get_signal(self, protocol: Protocol, name: str):
+        return self._all_signals[protocol][name]
+
 
-@dataclass
-class IOC:
-    process: subprocess.Popen
-    protocol: Literal["ca", "pva"]
+@pytest.fixture
+async def device_test_records() -> DeviceTestRecordsGroup:
+    device = DeviceTestRecordsGroup(PV_PREFIX)
+    await device.connect()
+    return device
 
-    async def make_backend(
-        self, typ: type | None, suff: str, timeout=10.0
-    ) -> SignalBackend:
-        # Calculate the pv
-        pv = f"{self.protocol}://{PV_PREFIX}:{self.protocol}:{suff}"
-        # Make and connect the backend
-        backend = _epics_signal_backend(typ, pv, pv)
-        await backend.connect(timeout=timeout)
-        return backend
+
+async def _make_backend(typ: type | None, protocol: str, suff: str, timeout=10.0):
+    pv = f"{protocol}://{PV_PREFIX}:{protocol}:{suff}"
+    # Make and connect the backend
+    backend = _epics_signal_backend(typ, pv, pv)
+    await backend.connect(timeout=timeout)
+    return backend
 
 
 # Use a module level fixture per protocol so it's fast to run tests. This means
 # we need to add a record for every PV that we will modify in tests to stop
 # tests interfering with each other
-@pytest.fixture(scope="module", params=["pva", "ca"])
-def ioc(request: pytest.FixtureRequest):
-    protocol = request.param
+@pytest.fixture(scope="module")
+def ioc(request: pytest.FixtureRequest, tmpdir_factory) -> subprocess.Popen:
+    with open(CA_PVA_RECORDS) as f:
+        ca_pva_template = f.read()
+    with open(PVA_RECORDS) as f:
+        pva_template = f.read()
+
+    template = ca_pva_template.replace("$(R)", "ca:")
+    template += ca_pva_template.replace("$(R)", "pva:")
+    template += pva_template.replace("$(R)", "pva:")
```

Comment:
We can pass many `-m` and `-d` flags to build up the database, like in https://github.com/bluesky/ophyd-async/blob/7d6070b30dc44e77bfd3c7024642460a2dffd27a/src/ophyd_async/epics/demo/__init__.py#L36-L38, rather than having to create a new `ALL_RECORDS` file...

Reply:
Come to think of it, we should try and work out a nice way of starting up the IOC subprocess for the `epics.demo` and `epics.testing` packages as a module level fixture here, and in the docs. Will discuss this with you in person...